core/src/main/scala/org/apache/spark/util/collection/ExternalSorter.scala

```diff
@@ -724,28 +653,7 @@ private[spark] class ExternalSorter[K, V, C](
     // Track location of each range in the output file
     val lengths = new Array[Long](numPartitions)
 
-    if (bypassMergeSort && partitionWriters != null) {
-      // We decided to write separate files for each partition, so just concatenate them. To keep
-      // this simple we spill out the current in-memory collection so that everything is in files.
-      spillToPartitionFiles(if (aggregator.isDefined) map else buffer)
```

Comment:
I think that this call was effectively a no-op since `buffer` should now be empty when `bypassMergeSort = true`.